src/audit-log-portal/src/utils/useGetMessage.ts

```diff
@@ -0,0 +1,26 @@
+import useSWR, { useSWRConfig } from "swr"
+import type { AuditLog } from "shared-types"
+import fetcher from "./fetcher"
+
+interface MessagesResult {
+  message: AuditLog
+}
+
+interface UseGetMessageResult {
+  message: AuditLog
+  error: Error
+  isLoading: boolean
+  reload: () => Promise<AuditLog[]>
+}
+
+export default function useGetMessages(url: string): UseGetMessageResult {
+  const { data, error } = useSWR(url, (fetchUrl: string) =>
+    fetcher<MessagesResult>(fetchUrl).then((result) => result.message)
+  )
+  const { mutate } = useSWRConfig()
+
+  const isLoading = !error && data === undefined
```

Comment:
is it possible that data can be null?

Reply:
yes, when looking up a non-existant message ID the api returns `{ message: null }`, and we get `null` instead of undefined here. Checking for whether `data` is `null` here led to the loading screen always being displayed when you searched for a message that didn't exist, and that case is handled by the conditional checks on `message `in https://github.com/ministryofjustice/bichard7-next-audit-logging/blob/9ed146e38f0ddb9d5bc30d7e939ebaa37e30d1ca/src/audit-log-portal/src/components/SearchResults/GetMessageByIdResults.tsx#L18